pytorch_lightning/trainer/trainer.py

```diff
@@ -799,6 +808,9 @@ def _run(self, model: LightningModule) -> Optional[Union[_EVALUATE_OUTPUT, _PRED
     def _pre_dispatch(self):
         self.accelerator.pre_dispatch(self)
 
+        # restore optimizers, etc.
+        self.checkpoint_connector.restore_training_state()
```

Comment:
Why is this here and not in `_run` as the rest of the `checkpoint_connector` calls?

Reply:
moved to _run() !